tests/sample/items/test_gradient_layer.py

```diff
@@ -0,0 +1,218 @@
+from unittest.mock import MagicMock
+
+import pytest
+from numpy.testing import assert_almost_equal
+
+import EasyReflectometry.sample.items.gradient_layer
+from EasyReflectometry.sample.item import GradientLayer
+from EasyReflectometry.sample.items.gradient_layer import (
+    _apply_thickness_constraints,
+    _linear_gradient,
+    _prepare_gradient_layers,
+)
+from EasyReflectometry.sample.layer import Layer
+from EasyReflectometry.sample.material import Material
+
+
+class TestGradientLayer():
+
+    @pytest.fixture
+    def gradient_layer(self):
+        self.init = Material.from_pars(10.0, -10.0, 'Material_1')
+        self.final = Material.from_pars(0.0, 0.0, 'Material_2')
+
+        return GradientLayer(
+            initial_material=self.init,
+            final_material=self.final,
+            thickness=1.0,
+            roughness=2.0,
+            discretisation_elements=10,
+            name='Test',
+            interface=None
+        )
+
+    def test_init(self, gradient_layer):
+        # When Then Expect
+        assert len(gradient_layer.layers) == 10 
+        assert gradient_layer.name, 'Test'
+        assert gradient_layer.type, 'Gradient-layer'
+        assert gradient_layer.interface is None
+        assert gradient_layer.thickness == 1.0
+        assert gradient_layer.layers.name == '0/1/2/3/4/5/6/7/8/9'
+        assert gradient_layer.layers[0].material.sld.raw_value == 10.0
+        assert gradient_layer.layers[5].material.sld.raw_value == 5.0
+        assert gradient_layer.layers[0].thickness.raw_value == 0.1
+        assert gradient_layer.layers[5].material.isld.raw_value == -5.0
+        assert gradient_layer.layers[9].material.isld.raw_value == -1.0
+
+    def test_default(self):
+        # When Then
+        result = GradientLayer.default()
+        
+        # Expect
+        assert result.name == 'Air-Deuterium'
+        assert result.type, 'Gradient-layer'
+        assert result.interface is None
+        assert len(result.layers) == 10
+        assert result.layers.name == '0/1/2/3/4/5/6/7/8/9'
+
+    def test_from_pars(self):
+        # When
+        init = Material.from_pars(6.908, -0.278, 'Boron')
+        final = Material.from_pars(0.487, 0.000, 'Potassium')
+        
+        # Then
+        result = GradientLayer.from_pars(
+            initial_material=init,
+            final_material=final,
+            thickness=10.0,
+            roughness=1.0,
+            discretisation_elements=5,
+            name='gradientItem'
+        )
+
+        # Expect
+        assert result.name, 'gradientItem'
+        assert result.type, 'Gradient-layer'
+        assert result.interface is None
+        assert len(result.layers) == 5
+        assert result.layers.name == '0/1/2/3/4'
+
+    def test_add_layer(self, gradient_layer):
+        # When Then Expect
```

Comment:
Your `When Then Expect` sounds more intuitive than the standard `Given When Expect`. 👌